mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list.go

```diff
@@ -0,0 +1,415 @@
+	if !ok {
+		resp.Diagnostics.AddError("error", errorAccessListCreate)
+		return
+	}
+
+	projectIPAccessListNewModel := newTFProjectIPAccessListModel(projectIPAccessListModel, entry)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &projectIPAccessListNewModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func newTFProjectIPAccessListModel(projectIPAccessListModel *tfProjectIPAccessListModel, projectIPAccessList *matlas.ProjectIPAccessList) *tfProjectIPAccessListModel {
+	entry := projectIPAccessList.IPAddress
+	if projectIPAccessList.CIDRBlock != "" {
+		entry = projectIPAccessList.CIDRBlock
+	} else if projectIPAccessList.AwsSecurityGroup != "" {
+		entry = projectIPAccessList.AwsSecurityGroup
+	}
+
+	id := fmt.Sprintf("%s-%s", projectIPAccessList.GroupID, entry)
+	return &tfProjectIPAccessListModel{
+		ID:               types.StringValue(id),
+		ProjectID:        types.StringValue(projectIPAccessList.GroupID),
+		CIDRBlock:        types.StringValue(projectIPAccessList.CIDRBlock),
+		IPAddress:        types.StringValue(projectIPAccessList.IPAddress),
+		AWSSecurityGroup: types.StringValue(projectIPAccessList.AwsSecurityGroup),
+		Comment:          types.StringValue(projectIPAccessList.Comment),
+		Timeouts:         projectIPAccessListModel.Timeouts,
+	}
+}
+
+func newMongoDBProjectIPAccessList(projectIPAccessListModel *tfProjectIPAccessListModel) []*matlas.ProjectIPAccessList {
+	return []*matlas.ProjectIPAccessList{
+		{
+			AwsSecurityGroup: projectIPAccessListModel.AWSSecurityGroup.ValueString(),
+			CIDRBlock:        projectIPAccessListModel.CIDRBlock.ValueString(),
+			IPAddress:        projectIPAccessListModel.IPAddress.ValueString(),
+			Comment:          projectIPAccessListModel.Comment.ValueString(),
+		},
+	}
+}
+
+func (r *ProjectIPAccessListRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var projectIPAccessListModelState *tfProjectIPAccessListModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &projectIPAccessListModelState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	entry := projectIPAccessListModelState.CIDRBlock.ValueString()
+	if projectIPAccessListModelState.IPAddress.ValueString() != "" {
+		entry = projectIPAccessListModelState.IPAddress.ValueString()
+	} else if projectIPAccessListModelState.AWSSecurityGroup.ValueString() != "" {
+		entry = projectIPAccessListModelState.AWSSecurityGroup.ValueString()
+	}
+
+	projectID := projectIPAccessListModelState.ProjectID.ValueString()
+	// Use the ID only with the IMPORT operation
+	var err error
+	if entry == "" && projectIPAccessListModelState.ID.ValueString() != "" {
+		projectID, entry, err = splitProjectIPAccessListImportID(projectIPAccessListModelState.ID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("the provided ID is not correct", err.Error())
+		}
+	}
+
+	timeout, diags := projectIPAccessListModelState.Timeouts.Read(ctx, projectIPAccessListTimeout)
```

Comment:
The read timeout is being set with a default of 45 minutes, while in previous implementation is was set to 2 minutes right?

Reply:
oh, thanks. Let me check. I will change it back to 2 minutes if that is the case

Reply:
Looking at this piece of code https://github.com/mongodb/terraform-provider-mongodbatlas/blob/af8d75bc7f81d2073931e776cba3fa9059cf2d5d/mongodbatlas/resource_mongodbatlas_project_ip_access_list.go#L91 I see that the default for the READ was 45 minutes. Where do you see it set to 2 minutes?

Reply:
I was actually looking at this value which seems to avoid the configurable timeout :/
https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_project_ip_access_list.go#L183

Reply:
mh... I am not sure if we should use 45 or 2 now 🤔 I will set it to two since we always used that value without considering the default one

Reply:
updated